contrib/babelfishpg_tds/src/backend/tds/tdsbulkload.c

```diff
@@ -1054,7 +1123,7 @@ ProcessBCPRequest(TDSRequest request)
 		pltsql_plugin_handler_ptr->stmt_needs_logging = false;
 		error_context_stack = plerrcontext;
 	}
-	offset = 0;
+	bcpOffset = 0;
```

Comment:
It's wrong to reset it here. What if we hit any of the previous two errors. We need to reset it immediately after the while loop. 
We should move the while loop inside the try catch and reset bcpOffset in both catch and end of try.
I don't see a need of introducing the reset function in that case.

```
PG_TRY();
{
    while() {}
}
PG_CATCH();
{
    bcpOffset = 0;
}
PG_END_TRY();
bcpOffset = 0;
```